Add-ons/UmbracoForms/Developer/Extending/Adding-a-Fieldtype.md

```diff
@@ -160,3 +160,106 @@ To reference the file you should override the `RenderView` property, e.g.:
+
+For example a Compare field type to compare another field.
+
+```csharp
+public class CompareField : Umbraco.Forms.Core.FieldType
+{
+    public CompareField()
+    {
+        this.Id = new Guid("b83dddc2-bdf3-4a0b-b9ad-f7bba83080df");
+        this.Name = "Compare Field";
+        this.Description = "Compare input to another field.";
+        this.Icon = "icon-autofill";
+        this.FieldTypeViewName = "FieldType.Compare.cshtml";
+        this.DataType = FieldDataType.String;
+        this.SortOrder = 10;
+        this.SupportsRegex = true;
+    }
+
+    [Setting("Field to compare",
+        Description = "Alias of field to compare.",
+        View = "textfield")]
+    public string FieldToCompare { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether the field label should be shown.
+    /// PreValues are a single element, a boolean indicating whether the default for the the checkbox is "checked".
+    /// </summary>
+    [Setting("Show Label", Description = "Indicate whether the the field's label should be shown when rendering the form.", View = "checkbox", PreValues = "true", DisplayOrder = 20)]
+    public string ShowLabel { get; set; } = string.Empty;
+
+    /// <inheritdoc/>
+    public override bool HideLabel => ShowLabel == "False"; // Checking explicitly for false so the backward compatible default is to show the label.
+
+    public override string GetDesignView() =>
+        "~/App_Plugins/UmbracoForms/backoffice/Common/FieldTypes/Textfield.html";
+
+    public override IEnumerable<string> ValidateField(Form form, Field field, IEnumerable<object> postedValues, HttpContext context, IPlaceholderParsingService placeholderParsingService)
+    {
+        var baseValidation = base.ValidateField(form, field, postedValues, context, placeholderParsingService);
+        var value = postedValues.FirstOrDefault();
+
+        var fieldToCompare = GetPostFieldValue(form, context, FieldToCompare);
+
+        if (fieldToCompare == null)
+            return baseValidation;
+
+        if (value != null && value.ToString() == fieldToCompare)
+        {
+            return baseValidation;
+        }
+
+        var custom = new List<string>();
+        custom.AddRange(baseValidation);
+        custom.Add("Not equal to.");
+
+        return custom;
+    }
+
+    private static string GetPostFieldValue(Form form, HttpContext context, string key)
+    {
+        Field? field = GetPostField(form, key);
+        if (field == null)
+        {
+            return string.Empty;
+        }
+
+        return context.Request.HasFormContentType && context.Request.Form.Keys.Contains(field.Id.ToString())
+            ? context.Request.Form[field.Id.ToString()].ToString().Trim()
+            : string.Empty;
+    }
+
+    private static Field? GetPostField(Form form, string key) => form.AllFields.SingleOrDefault(f => f.Alias == key);
```

Comment:
@AndyButland I noticed these method in other examples, but maybe Forms should include these extensions methods?

E.g.
```
GetField(Form form, string key)
GetFieldValue(Form form, HttpContext context, string key)
```